File: back/src/main/java/csh/back/domain/trip/place/dto/response/TripPlaceSaveResponse.java
package csh.back.domain.trip.place.dto.response;

import csh.back.domain.trip.place.entity.TripPlace;

public record TripPlaceSaveResponse(Long id,
                                    String name,
                                    String category,
                                    String address
    ) {
        public static TripPlaceSaveResponse from(TripPlace tripPlace) {
            return new TripPlaceSaveResponse(
                    tripPlace.getId(),
                    tripPlace.getName(),
                    tripPlace.getTheme(),
                    tripPlace.getAddress()
            );
        }
    }

File: back/src/main/kotlin/csh/back/domain/trip/place/dto/response/TripPlaceSaveResponse.kt
package csh.back.domain.trip.place.dto.response

import csh.back.domain.trip.place.entity.TripPlace

data class TripPlaceSaveResponse(
    val id: Long,
    val name: String,
    val category: String,
    val address: String,
) {
    companion object {
        @JvmStatic
        fun from(tripPlace: TripPlace): TripPlaceSaveResponse = TripPlaceSaveResponse(
            id = requireNotNull(tripPlace.id),
            name = tripPlace.name,
            category = tripPlace.category,
            address = tripPlace.address,
        )
    }
}